DonjinKrawler-Server/src/main/java/donjinkrawler/enemy_generation/EnemyGenerator.java

```diff
@@ -0,0 +1,52 @@
+package donjinkrawler.enemy_generation;
+
+import java.util.Random;
+
+public final class EnemyGenerator {
+
+    /**
+     * List of all available types of enemies
+     */
+    private static final String[] TYPES = {
+            "CHICKEN", "SKELETON", "ZOMBIE",
+    };
+
+    /**
+     * Enemy factory
+     */
+    private final EnemyFactory enemyFactory;
+
+    /**
+     * EnemyGenerator class Constructor
+     */
+    public EnemyGenerator() {
+        this.enemyFactory = new EnemyFactory();
+    }
+
+    /**
+     * Generate new random enemy
+     *
+     * @return new randomly generated enemy
+     */
+    public Enemy generateRandomEnemy() {
+        int rnd = new Random().nextInt(TYPES.length);
+        return this.enemyFactory.make(TYPES[rnd]);
+    }
+
+    /**
+     * Generate new random enemies
+     *
+     * @return new randomly generated enemies list
+     */
+    public Enemy[] generateRandomEnemies(int number) {
```

Comment:
Any particular reason you chose to use arrays instead of a List? ArrayList is the simplest implementation and you won't need to worry about indexes or making the list bigger.